components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/dao/BPSProfileDAO.java

```diff
@@ -0,0 +1,169 @@
+
+import org.apache.log4j.Logger;
+import org.wso2.carbon.identity.keyrotation.config.KeyRotationConfig;
+import org.wso2.carbon.identity.keyrotation.model.BPSPassword;
+import org.wso2.carbon.identity.keyrotation.util.KeyRotationException;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class holds implementations needed to re-encrypt the BPS Profile data in DB.
+ */
+public class BPSProfileDAO {
+
+    private static final Logger log = Logger.getLogger(BPSProfileDAO.class);
+    private static final BPSProfileDAO instance = new BPSProfileDAO();
+    private static final String PROFILE_NAME = "PROFILE_NAME";
+    private static final String USERNAME = "USERNAME";
+    private static final String TENANT_ID = "TENANT_ID";
+    private static final String PASSWORD = "PASSWORD";
+    public static int updateCount = 0;
+    public static int failedUpdateCount = 0;
+
+    public BPSProfileDAO() {
+
+    }
+
+    public static BPSProfileDAO getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * To retrieve the list of data in WF_BPS_PROFILE as chunks.
+     *
+     * @param startIndex        The start index of the data chunk.
+     * @param keyRotationConfig Configuration data needed to perform the task.
+     * @return List comprising of the records in the table.
+     * @throws KeyRotationException Exception thrown while retrieving data from WF_BPS_PROFILE.
+     */
+    public List<BPSPassword> getBpsPasswordChunks(int startIndex, KeyRotationConfig keyRotationConfig) throws
+            KeyRotationException {
+
+        List<BPSPassword> bpsPasswordList = new ArrayList<>();
+        String query = DBConstants.GET_BPS_PASSWORD;
+        int firstIndex = startIndex;
+        int secIndex = DBConstants.CHUNK_SIZE;
+        try (Connection connection = DriverManager
+                .getConnection(keyRotationConfig.getNewIdnDBUrl(), keyRotationConfig.getNewIdnUsername(),
+                        keyRotationConfig.getNewIdnPassword())) {
+            if (connection.getMetaData().getDriverName().contains(DBConstants.POSTGRESQL)) {
+                query = DBConstants.GET_BPS_PASSWORD_POSTGRE;
+                firstIndex = DBConstants.CHUNK_SIZE;
+                secIndex = startIndex;
+            } else if (connection.getMetaData().getDriverName().contains(DBConstants.MSSQL) ||
+                    connection.getMetaData().getDriverName().contains(DBConstants.ORACLE)) {
+                query = DBConstants.GET_BPS_PASSWORD_OTHER;
+            }
+            try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+                preparedStatement.setInt(1, firstIndex);
+                preparedStatement.setInt(2, secIndex);
+                ResultSet resultSet = preparedStatement.executeQuery();
```

Comment:
We need commit and rollback (for postres)

Reply:
Addressed with https://github.com/wso2/identity-tools/pull/13/commits/2c709c0658bfeb105b1836765c424d5ac02d18f5